extensions/twitch/src/helpers/auth.tsx

```diff
@@ -0,0 +1,109 @@
+import { getPreferenceValues, OAuth } from "@raycast/api";
+import fetch from "node-fetch";
+import { randomUUID } from "node:crypto";
+
+const preferences = getPreferenceValues<ExtensionPreferences>();
+
+export const clientId = preferences.clientId;
+const clientSecret = preferences.clientSecret;
+
+let runningAuthPromise: Promise<string> | undefined;
+
+export async function getHeaders() {
+  const promise = (runningAuthPromise ??= authorize());
+  promise.then(() => {
+    if (runningAuthPromise === promise) runningAuthPromise = undefined;
+  });
+  const accessToken = await promise;
+  const headers = {
+    "Client-Id": clientId,
+    Authorization: `Bearer ${accessToken}`,
+  } as const;
+  return headers;
+}
+
+const client = new OAuth.PKCEClient({
+  redirectMethod: OAuth.RedirectMethod.AppURI,
+  providerName: "Twitch",
+  providerIcon: "TwitchGlitchPurple.png",
+  description: "Connect your Twitch account…",
+  providerId: "twitch",
+});
+
+// Authorization
+
+async function authorize(): Promise<string> {
+  const tokenSet = await client.getTokens();
+  if (tokenSet?.accessToken) {
+    if (tokenSet.refreshToken && tokenSet.isExpired()) {
+      const newTokenSet = await refreshTokens(tokenSet.refreshToken);
+      await client.setTokens(newTokenSet);
+      return newTokenSet.access_token;
+    }
+    return tokenSet.accessToken;
+  }
+  const nonce = randomUUID();
+  const authRequest = await client.authorizationRequest({
+    endpoint: "https://id.twitch.tv/oauth2/authorize",
+    clientId,
+    scope: "user:read:follows",
+    extraParameters: {
+      response_type: "code",
+      redirect_uri: "https://raycast.com/redirect?packageName=Extension",
```

Comment:
These two parameters are automatically added by the `authorizationRequest` call. You shouldn't need to add these.